File: programmers/프린터.py
def solution(priorities, location):
    answer = 0

    result = []
    new_priorities = []
    
    for i in range(len(priorities)) :
        new_priorities.append((i,priorities[i]))
    
    while len(new_priorities) > 0 :

        temp = new_priorities.pop(0)
        isBool = False
        for j in range(0, len(new_priorities)) :
            if temp[1] < new_priorities[j][1] :
                isBool = True
                break
        if isBool :
            new_priorities.append(temp)
        else :
            result.append(temp)

    for i in range(len(result)) :

        if result[i][0] == location :
            answer = i+1
            break
    return answer

File: programmers/코테대비(2022.01.09)/프린터.java
import java.util.*;
class Point{
    int y;
    int x;
}

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        ArrayList<Point> arr = new ArrayList();
        
        for(int i=0;i<priorities.length;i++){
            Point point = new Point();
            point.y = priorities[i];
            point.x = i;
            arr.add(point);
        }
        int count=0;
        while(arr.size() > 0){
            Point pt = arr.remove(0);
            boolean check = true;
            for(int i=0;i<arr.size();i++){
                if(pt.y<arr.get(i).y){
                    arr.add(pt);
                    check = false;
                    break;
                }
            }
            if(check){

                if(pt.x == location){
                    answer = count+1;
                }
                count+=1;

            }
        }
        return answer;
    }
}
